wgpu-core/src/resource.rs

```diff
@@ -1967,8 +1934,14 @@ impl Drop for Tlas {
 }
 
 impl AccelerationStructure for Tlas {
-    fn raw<'a>(&'a self, guard: &'a SnatchGuard) -> Option<&'a dyn hal::DynAccelerationStructure> {
-        Some(self.raw.get(guard)?.as_ref())
+    fn try_raw<'a>(
+        &'a self,
+        guard: &'a SnatchGuard,
+    ) -> Result<&'a dyn hal::DynAccelerationStructure, DestroyedResourceError> {
+        self.raw
+            .get(guard)
+            .map(|raw| raw.as_ref())
+            .ok_or_else(|| DestroyedResourceError(self.error_ident()))
```

Comment:
What you wrote is completely fine, but I wonder if you think the below is slightly quicker to read:

```suggestion
        let Some(tlas) = self.raw.get(guard) else {
            return Err(DestroyedResourceError(self.error_ident()));
        };

        Ok(tlas.as_ref())
```

Reply:
I'm impartial to changing this but I copied it over from one of the other `try_raw` methods. I think if we want to change it we should change the others as well.